src/main/webapp/app/overview/course-dashboard/course-dashboard-guard.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, Router } from '@angular/router';
+
+import { first, lastValueFrom, of, switchMap } from 'rxjs';
+import { FeatureToggle, FeatureToggleService } from 'app/shared/feature-toggle/feature-toggle.service';
+import { CourseStorageService } from 'app/course/manage/course-storage.service';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class CourseDashboardGuard implements CanActivate {
+    constructor(
+        private featureToggleService: FeatureToggleService,
+        private courseStorageService: CourseStorageService,
+        private router: Router,
+    ) {}
+
+    /**
+     * Check if the client can activate a route.
+     * @return true if CourseD is enabled for this instance, false otherwise
+     */
+    canActivate(route: ActivatedRouteSnapshot): Promise<boolean> {
+        return lastValueFrom(
+            this.featureToggleService.getFeatureToggleActive(FeatureToggle.StudentCourseAnalyticsDashboard).pipe(
+                first(),
+                switchMap((isActive) => {
+                    const courseId = route.parent?.paramMap.get('courseId');
+                    if (!courseId) {
+                        return of(false);
+                    }
+
+                    if (isActive) {
+                        // Check if course has dashboard enabled
+                        const course = this.courseStorageService.getCourse(parseInt(courseId, 10));
```

Comment:
The course of `getCourse` is `undefined` if it was not stored before. On refresh on the dashboard site the course is not immediately fetched, that is why it redirects to `/exercises`